### PR TITLE
Add Repository for kuksa-java-sdk

### PR DESCRIPTION
**Pull request by wba2hi**

### CODE REVIEW

**Comment by wba2hi** [comment]
Anything I have to do / can do here: https://github.com/eclipse-kuksa/.eclipsefdn/pull/33#issuecomment-2536140136?

**Comment by netomi** [comment]
should be fine now.

**Comment by wba2hi** [comment]
Thanks for your (always) quick support and help @netomi :)

**Comment by netomi** [comment]
np, environments are annoying, as they are kept being auto-created by GitHub. We should probably remove them from the config to avoid problems with them. It is useful sometimes to be able to configure them though.

File: otterdog/eclipse-kuksa.jsonnet
Comment (lukasmittag) [inline otterdog/eclipse-kuksa.jsonnet:419]: `description: "KUKSA performance tests",` description like above missing please add

**Review by lukasmittag — CHANGES_REQUESTED** [review]

File: otterdog/eclipse-kuksa.jsonnet
Comment (erikbosch) [inline otterdog/eclipse-kuksa.jsonnet:422]: Here we use the same merge strategy as for the android-repos, i.e. only allow merge-commits, do not allow rebase/squash. As I understood there were some reasons to have that setup for Android, but does those reasons apply here as well, or do we want to have same setup as for most other repos where we allow all three methods?

File: otterdog/eclipse-kuksa.jsonnet
Comment (wba2hi) [reply otterdog/eclipse-kuksa.jsonnet:419]: Description has been added

**Comment by SebastianSchildt** [comment]
In case it needs an approve from me before merge, I will wait for @erikbosch approval and then follow up if I like what I see

File: otterdog/eclipse-kuksa.jsonnet
Comment (Chrylo) [reply otterdog/eclipse-kuksa.jsonnet:422]: I don't mind changing the approach here. I would like some kind of consistency though. The approach here is based on [GitLab Flow](https://docs.gitlab.co.jp/ee/topics/gitlab_flow.html). Are the other repos based on some official branching flow? 

However there are some disadvantages of squashing with the current commit approach (conventional commits). The commits are linted for a consistency. Automatic Squash Merges will reduce the usefulness of this. So I am against that one. However you can squash it manually and force push before the merge if you want.

We can switch from merge commits to always rebase here if that feels better.

Comment (lukasmittag) [reply otterdog/eclipse-kuksa.jsonnet:422]: would tend to rebase merge as well

Comment (erikbosch) [reply otterdog/eclipse-kuksa.jsonnet:422]: We do not have any official branching flow in other repos. Typically there seems to be two sorts of people, those who prefer a clean history and [fast forward merge](https://docs.gitlab.co.jp/ee/user/project/merge_requests/methods/index.html#fast-forward-merge), and those who prefer merge-commits. So in the end regardless of policy we will annoy some people. Personally I prefer a clean history unless the source branch is something long-lived, like merging "main" to "production" if you use the GitLab flow

There are also different views on how to keeping feature branches up-to-date, some prefer rebase and force-push to keep linear history, other prefer merges to make sure that there are no "orphan" comments. The general risk I see if you use merge-commits (or rebase) is that contributors will not do a rebase/squash/force-push to the feature branch before merge, so you end up with a lot of in my view use-less commit messages like "Applying suggestions from code-review"

The problem with too aggressive commit linters is that you might get errors also for commits created by bots like in https://github.com/eclipse-kuksa/kuksa-android-sdk/pull/119, see https://github.com/eclipse-kuksa/kuksa-android-sdk/actions/runs/12070583813, as dependabot does not follow "subject must be sentence-case"

**Review by erikbosch — APPROVED** [review]

**Review by SebastianSchildt — APPROVED** [review]
lgtm 👍

File: otterdog/eclipse-kuksa.jsonnet
Comment (Chrylo) [reply otterdog/eclipse-kuksa.jsonnet:422]: Always a hot topic because the pros and contras are difficult to clearly compare.^^

I suggest:
- Doing rebase for the clean history so it is aligned with most other repos
- Keep using conventional commit and automatic changelog / release note generation. The advantage is too big IMHO. 
- Do not allow merge commits for consistency. Either rebase or merge commits.
- Do not allow automatic squashing because it should be done deliberately to ensure a good commit quality for the conventional commits standard. 
- Document a clear Branching / Release concept (Preferable taking an official one) so all the rules are clearly linked and can be easily copied to a new repo. 

If we don't care about the commit history or automatic changelog generation then we can think about allowing "everything". Could be that the advantages are not worth the trade off. But that is currently the basis for my arguments. :)

The risk that OSS Contributors are restricted too much by linters etc. to contribute is always there with every quality gate we are introducing. Be it code quality or commit quality tools. My feeling is that a good commit quality is ignored too much. That is why the main contributor Role is so important. You probably never want to just merge a (bigger) PR from outside without cleaning it up.

We should probably also not enforce "subject must be sentence-case" since dependabot is already using the conventional commit standards OR change the config for the dependabot for the commit message (if possible).

@erikbosch I let you decide together with Andre as the main reviewer  one here without hard feelings.  My motivation is only to solve the "bad commit messages" issue and automate release notes generation.

**Review by Chrylo — APPROVED** [review]
LGTM

**Pull request merged**
